modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/action/executor/test/ObjectActionExecutorRegistryImplTest.java

```diff
@@ -0,0 +1,144 @@
+import org.osgi.framework.ServiceRegistration;
+
+/**
+ * @author Guilherme Camacho
+ */
+@RunWith(Arquillian.class)
+public class ObjectActionExecutorRegistryImplTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testShouldNotReturnObjectActionExecutorScopedByAnotherCompany() {
+		ObjectActionExecutor testObjectActionExecutor1 =
+			new TestObjectActionExecutor(1, ListUtil.fromString("abc"));
+
+		ObjectActionExecutor testObjectActionExecutor2 =
+			new TestObjectActionExecutor(2, ListUtil.fromString("abc"));
+
+		ServiceRegistration<ObjectActionExecutor>
+			objectActionExecutorServiceRegistration1 = null;
+
+		ServiceRegistration<ObjectActionExecutor>
+			objectActionExecutorServiceRegistration2 = null;
+
+		try {
+			objectActionExecutorServiceRegistration1 =
+				_registerObjectActionExecutor(testObjectActionExecutor1);
+
+			objectActionExecutorServiceRegistration2 =
+				_registerObjectActionExecutor(testObjectActionExecutor2);
+
+			List<ObjectActionExecutor> objectActionExecutors =
+				_objectActionExecutorRegistry.getObjectActionExecutors(
+					1, "abc");
+
+			Assert.assertFalse(
+				objectActionExecutors.contains(testObjectActionExecutor2));
+		}
+		finally {
+			_unregisterObjectActionExecutor(
+				objectActionExecutorServiceRegistration1);
+			_unregisterObjectActionExecutor(
+				objectActionExecutorServiceRegistration2);
+		}
+	}
+
+	private ServiceRegistration<ObjectActionExecutor>
+		_registerObjectActionExecutor(
+			ObjectActionExecutor objectActionExecutor) {
+
+		Bundle bundle = FrameworkUtil.getBundle(
+			ObjectActionExecutorRegistryImplTest.class);
+
+		BundleContext bundleContext = bundle.getBundleContext();
+
+		return bundleContext.registerService(
+			ObjectActionExecutor.class, objectActionExecutor, null);
+	}
+
+	private void _unregisterObjectActionExecutor(
+		ServiceRegistration<ObjectActionExecutor>
+			objectActionExecutorServiceRegistration) {
+
+		if (objectActionExecutorServiceRegistration == null) {
+			return;
+		}
+
+		objectActionExecutorServiceRegistration.unregister();
+	}
+
+	@Inject
+	private ObjectActionExecutorRegistry _objectActionExecutorRegistry;
+
+	private static class TestObjectActionExecutor
+		implements CompanyScoped, ObjectActionExecutor, ObjectDefinitionScoped {
+
+		public TestObjectActionExecutor(
+			long allowedCompanyId, List<String> allowedObjectDefinitionNames) {
+
+			_allowedCompanyId = allowedCompanyId;
+			_allowedObjectDefinitionNames = allowedObjectDefinitionNames;
+		}
+
+		@Override
+		public void execute(
+				long companyId, UnicodeProperties parametersUnicodeProperties,
+				JSONObject payloadJSONObject, long userId)
+			throws Exception {
+		}
+
+		@Override
+		public long getAllowedCompanyId() {
+			return _allowedCompanyId;
+		}
+
+		@Override
+		public List<String> getAllowedObjectDefinitionNames() {
+			return _allowedObjectDefinitionNames;
+		}
```

Comment:
Hey @MarinhoFeliphe , I prefer to keep it dynamic, because I want to use this same class to cover other test scenarios that involve allowedObjectDefinitionNames. For this test, I will pass `Collections.emptyList()` instead of `ListUtil.fromString("abc")`

Reply:
agreed!

> I want to use this same class to cover other test scenarios that involve allowedObjectDefinitionNames

@guilhermedcamacho I'm not sure, but the same way we don't need to pass the key `"test"` maybe Brian will ask for adapt this only when we will be writting this test about allowedObjectDefinitionNames, what do you think?

Thanks!